nemo/collections/common/parts/preprocessing/manifest.py

```diff
@@ -206,6 +207,12 @@ def get_full_path(
         ]
     elif isinstance(audio_file, str):
         # If input is a string, get the corresponding full path
+        if "/sharded_manifests/" in manifest_file or os.path.basename(manifest_file) == "tarred_audio_manifest.json":
+            logging.warning(
+                f"Manifest file `{manifest_file}` seems to be part of a tarred dataset, skip checking for relative paths. If this is not intended, please avoid having `/sharded_manifests/` and `tarred_audio_manifest.json` in manifest_filepath.",
+                mode=LogMode.ONCE,
+            )
+            return audio_file
```

Comment:
I agree that passing a flag from dataset to item_iter is the best way, but that would require modifying a lot of dataset classes that have tarred versions. I'll first add the `is_audio_from_tarred_dataset(audio_file, manifest_file)` as you suggest~

Reply:
why not just check for 'shard_id' key in parse_id and call/skip `get_full_path` function accordingly?

Reply:
> why not just check for shard_id key in parse_item function and call/skip get_full_path function accordingly?

it's easier to just change the `get_full_path` function, since there are many other calls to this function elsewhere